packages/contracts/test/core/permission/permission-manager.ts

```diff
@@ -791,13 +867,199 @@ describe('Core: PermissionManager', function () {
+      ).deploy();
+      const genericTargetCondition = await new PermissionConditionMock__factory(
+        signers[0]
+      ).deploy();
+
+      // Grant with a specific condition that will answer false
+      await pm.grantWithCondition(
+        pm.address,
+        ownerSigner.address,
+        ADMIN_PERMISSION_ID,
+        specificCondition.address
+      );
+      await specificCondition.setAnswer(false);
+
+      // Grant with a generic caller condition that will answer true
+      await pm.grantWithCondition(
+        pm.address,
+        ANY_ADDR,
+        ADMIN_PERMISSION_ID,
+        genericCallerCondition.address
+      );
+      await genericCallerCondition.setAnswer(true);
+
+      // Grant with a generic target condition that will answer true
+      await pm.grantWithCondition(
+        ANY_ADDR,
+        ownerSigner.address,
+        ADMIN_PERMISSION_ID,
+        genericTargetCondition.address
+      );
+      await genericCallerCondition.setAnswer(true);
+
+      // Check that `isGranted` returns false for `ownerSigner` to whom the specific condition was granted.
+      expect(
+        await pm.isGranted(
+          pm.address,
+          ownerSigner.address,
+          ADMIN_PERMISSION_ID,
+          genericTargetCondition.address
+        )
+      ).to.be.false;
+
+      // Check that `ownerSigner` is still granted access to other contracts (e.g., `address(0)`) through the `genericTargetCondition` condition.
+      expect(
+        await pm.isGranted(
+          ethers.constants.AddressZero,
+          ownerSigner.address,
+          ADMIN_PERMISSION_ID,
+          genericTargetCondition.address
+        )
+      ).to.be.true;
+    });
+
+    it('does not fall back to a generic target condition if a generic caller condition is set already answering `false`', async () => {
+      const genericCallerCondition = await new PermissionConditionMock__factory(
+        signers[0]
+      ).deploy();
+      const genericTargetCondition = await new PermissionConditionMock__factory(
+        signers[0]
+      ).deploy();
+
+      // Grant with a generic caller condition that will answer false.
+      await pm.grantWithCondition(
+        pm.address,
+        ANY_ADDR,
+        ADMIN_PERMISSION_ID,
+        genericCallerCondition.address
+      );
+      await genericCallerCondition.setAnswer(false);
+
+      // Grant with a generic target condition that will answer true.
+      await pm.grantWithCondition(
+        ANY_ADDR,
+        ownerSigner.address,
+        ADMIN_PERMISSION_ID,
+        genericTargetCondition.address
+      );
+      await genericTargetCondition.setAnswer(true);
+
+      // Check that `isGranted` returns false for `ANY_ADDR` (here, we check only two addresses, `ownerSigner` and `otherSigner`).
+      expect(
+        await pm.isGranted(
+          pm.address,
+          ownerSigner.address,
+          ADMIN_PERMISSION_ID,
+          genericTargetCondition.address
+        )
+      ).to.be.false;
+      //
```

Comment:
remove or add actual comment

Reply:
Fixed in f9aab5d09b636af7735c106c92f60408ee900d94